### PR TITLE
Merging `<App/ForumPage>` into the project and Integrate route with existing `<App/ChannelPage>`

### PR DESCRIPTION
**Pull request by NorrisWu0**
In short, this pr added working forum page and updated channel page so it kinda work. Still need to update the backend API (not surprisingly)

### What I have not done yet?
- [ ] Populate this pull request with useful information
- [ ] Leaving this for me tomorrow
- [ ] Update this
- [ ] Update this
- [ ] Update this
- [ ] 重要的事情说三遍

### CODE REVIEW

**Comment by NorrisWu0** [comment]
Okay, CICD failed because I only implemented very basic "error handling". Will fix this tomorrow.

File: src/pages/forum/ChannelPage/ChannelPage.jsx [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
+  const { channelId } = useParams();
+
+  const tableData = {
+    colGroup: ['0', '', '0', '0', '0'],
+    header: ['pinned', 'threads', 'comments', 'likes', 'last comment']
+  };
 
   useEffect(() => {
-    indexPost().then((res) => setPosts(res));
+    console.log(`This channel is ${channelId}`);
```
Comment (Ollie-lee) [inline src/pages/forum/ChannelPage/ChannelPage.jsx:19]: 没用的log在prod前记得删掉

File: src/pages/forum/ChannelPage/ChannelPage.jsx [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
+    indexPost()
+      .then((res) => setPosts(res))
+      .catch((err) => console.log(err));
```
Comment (Ollie-lee) [inline src/pages/forum/ChannelPage/ChannelPage.jsx:22]: 在前端,error最好要有UI展示,当然现在项目时间不够,不写也没事

File: src/pages/forum/ChannelPage/components/PostItem/PostItem.jsx [hunk 3]
```diff
@@ -19,7 +19,9 @@ export default function PostItem({ post }) {
       <td>{pinned && <FaThumbtack />}</td>
       <td className="post-item__thread">
         <h2 className="post-item__text--level-large">{title}</h2>
-        <p className="post-item__text--level-small">{author.username}</p>
+        <p className="post-item__text--level-small">
+          {author && author.username}
```
Comment (Ollie-lee) [inline src/pages/forum/ChannelPage/components/PostItem/PostItem.jsx:23]: 或者用es6的optional chaining 
```jsx
author?.username
```

File: src/pages/forum/ForumPage/ForumPage.jsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import './ForumPage.css';
+
+import PropTypes from 'prop-types';
+
+const tableData = {
+  colGroup: ['', '0', '0'],
+  header: ['channel', 'posts', 'last post']
+};
+
+const displayFormattedDate = (now) => {
```
Comment (Ollie-lee) [inline src/pages/forum/ForumPage/ForumPage.jsx:16]: 这种函数可以写个注释放到utils.js里面

File: src/pages/forum/ForumPage/ForumPage.jsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  const { url } = useRouteMatch();
+
+  if (!_.isEmpty(channels)) {
+    return channels.map(({ category, channel, lastPost, postsCount }) => {
+      const to = channel.toLowerCase().replace(' ', '-');
+      const post = useMemo(
+        () =>
+          _.isUndefined(lastPost)
+            ? ' '
+            : displayFormattedDate(lastPost.updatedAt),
```
Comment (Ollie-lee) [inline src/pages/forum/ForumPage/ForumPage.jsx:35]: 试试更优雅的,line33-35一行搞定
```jsx
displayFormattedDate?.(lastPost?.updatedAt) ?? "not available"
```

File: src/pages/forum/ForumPage/ForumPage.jsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  const tabs = useMemo(() => {
+    const newArr = [];
+    categories.forEach(({ _id }) => {
+      newArr.push(_id);
+    });
+    setCurrentTab(newArr[0]);
+    return newArr;
+  }, [categories]);
+
+  useEffect(() => {}, [tabs]);
```
Comment (Ollie-lee) [inline src/pages/forum/ForumPage/ForumPage.jsx:92]: 这行的作用是？

**Review by Ollie-lee — APPROVED** [review]

File: src/pages/forum/ForumPage/ForumPage.jsx (hunk 6 above)
Comment (NorrisWu0) [reply src/pages/forum/ForumPage/ForumPage.jsx:92]: 太匆忙忘删了😂😂 明早起来修正

File: src/pages/forum/ForumPage/ForumPage.jsx (hunk 5 above)
Comment (NorrisWu0) [reply src/pages/forum/ForumPage/ForumPage.jsx:35]: 这optional chaining也太6了 😮

**Pull request merged**